clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -645,9 +645,10 @@ offline_instru_t::insert_save_addr(void *drcontext, instrlist_t *ilist, instr_t
          * store the base reg directly and add the disp during post-processing.
          */
         reg_addr = opnd_get_base(ref);
-        if (opnd_get_base(ref) == reg_ptr) {
-            /* Here we do need a scratch reg, and raw2trace can't identify this case:
-             * so we set disp to 0 and use the regular path below.
+        if (opnd_get_base(ref) == reg_ptr || opnd_get_base(ref) == dr_get_stolen_reg()) {
```

Comment:
Sorry I had it in mind but forgot to ask before: is there a similar control point in raw2trace that should be aware of this? I know we share elision logic for that same reason between online tracer and raw2trace..

Reply:
Oh this doesn't change the count of emitted addresses, so no need for raw2trace adjustmnet.